### PR TITLE
Remove fallback to v1alpha1 authentication

### PR DESCRIPTION
**Pull request by diemtvu**
This is a first step to to deprecated v1alpha1 authentication policy.

### CODE REVIEW

**Comment by incfly** [comment]
some e2e test supposed to fail as expected

**Comment by diemtvu** [comment]
> some e2e test supposed to fail as expected?

Right. It's easier to know which tests need to be updated/removed :)

**Comment by diemtvu** [comment]
All tests are fix. @yangminzhu @incfly @howardjohn , PTAL. Thanks.

File: tests/integration/pilot/cni/testdata/global-mtls-on.yaml [hunk 1]
```diff
@@ -1,10 +1,10 @@
-apiVersion: authentication.istio.io/v1alpha1
-kind: MeshPolicy
+apiVersion: "security.istio.io/v1beta1"
+kind: "PeerAuthentication"
```
Comment (hzxuzhonghu) [inline tests/integration/pilot/cni/testdata/global-mtls-on.yaml]: nit: quota is unnecessary

File: tests/integration/security/mtls_first_party_jwt/testdata/global-mtls-on-no-dr.yaml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 # mTLS is enabled in strict mode without destination rule.
-apiVersion: authentication.istio.io/v1alpha1
-kind: MeshPolicy
+apiVersion: "security.istio.io/v1beta1"
+kind: "PeerAuthentication"
```
Comment (hzxuzhonghu) [inline tests/integration/security/mtls_first_party_jwt/testdata/global-mtls-on-no-dr.yaml]: same

**Pull request merged**